gui/util/range_slider.py

```diff
@@ -73,7 +73,7 @@ def setValues(self, values):
             Start and end of the range.
         """
         self.scale_min, self.scale_max = values
-        self.emitRange()
+        #self.emitRange()
```

Comment:
any specific reason for this?

Reply:
what was happening was that when I was calling `climSliderUpdate` whenever the layer selection was changed, it would call `climSlider.setValues` to reposition the sliders which would trigger an `emitRange()` which would then call a `climSliderChanged`. Somehow when this loop was going on the slider position updates would cease to function as expected. For example if you added two layers, put the clims slider to different positions, turned the visibility off on one of them and switched between them in the layer widgets the update stopped working. For some reason toggling the visibility was affecting things which surprised me a lot. Commenting out this line fixed the strange behavior, but if we want to keep the line I can try and chase down where the problem was coming from

Reply:
Up to you. We could fix this now or in the refactor.

Reply:
i'll look into it - i take it from you though that having this `emitRange` get called is desirable. Is that your feeling too @AhmetCanSolak?

Reply:
Yeah, I want to be able to hook callbacks onto it.

Reply:
Yes @sofroniewn . it will be useful to hook.

Reply:
I'll add this back in, thanks!